AdvLoggerPkg/Library/AdvancedLoggerLib/MmCoreArm/AdvancedLoggerLib.c

```diff
@@ -102,48 +54,59 @@ AdvancedLoggerGetLoggerInfo (
   VOID
   )
 {
-  EFI_STATUS            Status;
-  EFI_PHYSICAL_ADDRESS  Address;
+  ADVANCED_LOGGER_INFO  *LoggerInfo;
+  EFI_PHYSICAL_ADDRESS  MaxAddress;
+  UINT32                HwPrintLevel;
 
   ASSERT (FeaturePcdGet (PcdAdvancedLoggerFixedInRAM));
   if (!FeaturePcdGet (PcdAdvancedLoggerFixedInRAM)) {
     return NULL;
   }
 
-  if (!mInitialized) {
-    // If this is the first time for MM core to get here, the memory attributes of this module
-    // may not be fully set yet. Thus set the memory for global variables attributes to RW first.
-    Address = ALIGN_VALUE ((EFI_PHYSICAL_ADDRESS)(UINTN)&mInitialized - EFI_PAGE_SIZE + 1, EFI_PAGE_SIZE);
-    Status  = MmuSetAttributes (Address, EFI_PAGE_SIZE, EFI_MEMORY_XP);
-    ASSERT_EFI_ERROR (Status);
-    Status = MmuClearAttributes (Address, EFI_PAGE_SIZE, EFI_MEMORY_RO);
-    ASSERT_EFI_ERROR (Status);
-
-    Address = ALIGN_VALUE ((EFI_PHYSICAL_ADDRESS)(UINTN)&mLoggerInfo - EFI_PAGE_SIZE + 1, EFI_PAGE_SIZE);
-    Status  = MmuSetAttributes (Address, EFI_PAGE_SIZE, EFI_MEMORY_XP);
-    ASSERT_EFI_ERROR (Status);
-    Status = MmuClearAttributes (Address, EFI_PAGE_SIZE, EFI_MEMORY_RO);
-    ASSERT_EFI_ERROR (Status);
-
-    mInitialized = TRUE;            // Only allow initialization once
-    mLoggerInfo  = (ADVANCED_LOGGER_INFO *)(VOID *)FixedPcdGet64 (PcdAdvancedLoggerBase);
-    ASSERT (mLoggerInfo != NULL);
-    if (mLoggerInfo == NULL) {
-      return NULL;
-    }
-
-    mLoggerInfo->HwPrintLevel = FixedPcdGet32 (PcdAdvancedLoggerHdwPortDebugPrintErrorLevel);
-
-    mMaxAddress = mLoggerInfo->LogBuffer + mLoggerInfo->LogBufferSize;
-    mBufferSize = mLoggerInfo->LogBufferSize;
+  LoggerInfo = (ADVANCED_LOGGER_INFO *)(VOID *)FixedPcdGet64 (PcdAdvancedLoggerBase);
+  if (LoggerInfo == NULL) {
+    return NULL;
+  }
+
+  // Initialize HdwPrintLevel if needed.
+  HwPrintLevel = FixedPcdGet32 (PcdAdvancedLoggerHdwPortDebugPrintErrorLevel);
+  if (LoggerInfo->HwPrintLevel != HwPrintLevel) {
+    LoggerInfo->HwPrintLevel = HwPrintLevel;

```

Comment:
Can the HwPrintLevel be overwritten by configuration code? There are some slightly complicated flows to configuring what is getting logged, I'm wondering if this could overwrite later configuration

Reply:
As discussed offline, I will remove this. Right now, this implementation only can handle a adv logger buffer that was previously intialized. As such it should not presume to set any initial values as it cannot know the origin of the existing value. Only the initial creator of the structure decides default values.